### PR TITLE
Fixes #460: Auto refreshing mode for media walls

### LINKED ISSUE #460: Auto-refreshing mode for media walls
The idea is to 

- [ ] Sent request to the loklak server after 10 seconds for new feeds and getting an auto-refreshing mode for media walls.

- [ ] Adding animations to add and display new feeds to media wall in an appealing way.

### PR DESCRIPTION
**Pull request by Achint08**
**Changes proposed in this pull request**
- Auto refreshing mode for media wall
- Adds Animation for adding new feeds on media wall
- Triggers new request after every 5 seconds
- Separates actions, reducers for media wall.

**Link to live demo (if appropriate):** 
https://achint08.github.io/loklak_search/

**Closes #460**

### CODE REVIEW

**Comment by Achint08** [comment]
Open for review @djmgit @hemantjadon @kavithaenair @singhpratyush @SKrPl  @vibhcool .

**Comment by vibhcool** [comment]
Is there an error? nothing appears on media wall?
screenshot from 2017-07-23 04-14-59

File: src/app/effects/media-wall-pagination.effects.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ * the payload of the currently dispatched action, useful in
+ * instances where the current state is not necessary.
+ *
+ * A simple way to think of it is that ngrx/effects is an event listener of sorts.
+ * It listens for actions being dispatched to the store. You can then tell `ngrx/effects`
+ * that when a particular action is dispatched, to take another, new action as a result.
+ * At the end, what’s really happening is `ngrx/effects` is an `action generator` that dispatches
+ * a `new action` as a result of a different action.
+ */
+
```
Comment (vibhcool) [inline src/app/effects/media-wall-pagination.effects.ts]: empty line here.

File: src/app/reducers/api-media-wall-response.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ * @prop: entities: []
+ * @prop: lastResponeLength: 0
+ */
+export const initialState: State = {
+	entities: [],
+	lastResponseLength: 0
+};
+
+
```
Comment (vibhcool) [inline src/app/reducers/api-media-wall-response.ts]: extra line

File: src/app/reducers/media-wall-search.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * There is always a need of initial state to be passed onto the store.
+ *
+ * @prop: loading: false
+ */
+export const initialState: State = {
+	loading: false
+};
+
+
```
Comment (vibhcool) [inline src/app/reducers/media-wall-search.ts]: a line space extra

File: src/app/reducers/media-wall-search.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * The actual reducer function. Reducers can be thought of as the tables in the DataBase.
+ * These are the functions which are responsible for maintaing, and updating the
+ * `State` of the application they control.
+ *
+ * Here the reducer cotrols that part of the state which is shows the state of the application
```
Comment (vibhcool) [inline src/app/reducers/media-wall-search.ts]: typo errors,  `cotrol` -> `control`

**Comment by vibhcool** [comment]
though this isn't related to issue, but wouldn't it be better if media wall opens in same tab?

File: src/app/reducers/media-wall-pagination.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * The actual reducer function. Reducers can be thought of as the tables in the DataBase.
+ * These are the functions which are responsible for maintaing, and updating the
+ * `State` of the application they control.
+ *
+ * Here the reducer controls the part of state which is responsilble for storing the
+ * results fetched from the API.
+ */
+export function reducer(state: State = initialState, action: wallPaginationAction.Actions): State {
```
Comment (vibhcool) [inline src/app/reducers/media-wall-pagination.ts]: It appears that this file and `src/app/reducers/media-wall-search.ts` can be merged and this method can also be merged.

Could you see if these can b merged :)

File: src/app/actions/media-wall.ts [hunk 6]
```diff
@@ -1,17 +1,35 @@
...
+ * See Discriminated Unions: https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions
+ */
 
-export class WallInputValueChangeAction implements Action {
-	type = ActionTypes.WALL_INPUT_VALUE_CHANGE;
+export class WallSearchAction implements Action {
+	type = ActionTypes.WALL_SEARCH;
 
-	constructor(public payload: string) { }
+	constructor(public payload: Query) {  }
```
Comment (vibhcool) [inline src/app/actions/media-wall.ts]: extra space here -> constructor(public payload: Query) `{  }`

**Comment by vibhcool** [comment]
also media wall refreshed once. didn't repeat the refresh. Shouldn't it refresh, repeatedly after an interval?

File: src/app/effects/media-wall-pagination.effects.ts (hunk 1 above)
Comment (Achint08) [inline src/app/effects/media-wall-pagination.effects.ts]: For user readibility.

File: src/app/reducers/api-media-wall-response.ts (hunk 2 above)
Comment (Achint08) [inline src/app/reducers/api-media-wall-response.ts]: User readibilty.

File: src/app/reducers/media-wall-search.ts (hunk 3 above)
Comment (Achint08) [inline src/app/reducers/media-wall-search.ts]: User readibilty.

File: src/app/reducers/media-wall-search.ts (hunk 4 above)
Comment (Achint08) [inline src/app/reducers/media-wall-search.ts]: Done. :)

File: src/app/reducers/media-wall-pagination.ts (hunk 5 above)
Comment (Achint08) [inline src/app/reducers/media-wall-pagination.ts]: These two files control different states.

**Comment by Achint08** [comment]
@vibhcool 
> though this isn't related to issue, but wouldn't it be better if media wall opens in same tab?

In the current deployment, media wall will open in the same tab.

> also media wall refreshed once. didn't repeat the refresh. Shouldn't it refresh, repeatedly after an interval?

If there are no feeds are received after a certain request, auto-refresh will stop.

Made some changes on spacing.  Please check.

**Comment by hemantjadon** [comment]
@Achint08 for media walls can the polling time be dynamic? I mean instead of completely stopping the fetching of results can we make it poll periodically after some time to check whether results are available or not?

**Comment by Achint08** [comment]
@hemantjadon In that case, results will be repeated.

**Comment by hemantjadon** [comment]
If we use pagination API start record and all?

**Comment by Achint08** [comment]
@hemantjadon Suppose new feeds are added after some time, then indexing will change, I think. Let me confirm it, I will add this in subsequent PR if this is not the case.

**Comment by hemantjadon** [comment]
@Achint08 Okay we can add if possible such functionality in some other PR

**Comment by vibhcool** [comment]
> If we use pagination API start record and all?

is it part of JS?

There can be another approach, get `created_at` time of last result and use `since` and `until` parameters

**Comment by Achint08** [comment]
> There can be another approach, get created_at time of last result and use since and until parameters.

@vibhcool I will look into this.

@hemantjadon @mariobehling Can this be merged. So that I can proceed with my next PRs.

**Comment by hemantjadon** [comment]
This looks good to me So merging this.

**Pull request merged**